SteamKit2/SteamKit2/Steam/Discovery/SmartCMServerList.cs

```diff
@@ -264,7 +233,8 @@ private IPEndPoint GetNextServerCandidateInternal()
                 ResetOldScores();
 
                 var serverInfo = servers
-                    .Select( (s, index) => new { EndPoint = s.EndPoint, IsBad = s.LastBadConnectionDateTimeUtc.HasValue, Index = index } )
+                    .Where( s => (s.Record.ProtocolTypes & supportedProtocolTypes) > 0 )
```

Comment:
You're already using LINQ, `HasFlag` has negligible performance hit. Sure there is type check and boxing/unboxing (everything is casted to `ulong` IIRC), but this is still negligible compared to huge advantage of clean code. This is just my personal opinion though.